### ISSUE TITLE
Feature request: show container names

### ISSUE THREAD

**Issue by mudasar187** [jesseduffield/lazydocker#40]
Hi, i'v tested out lazydocker today and i love it!
A feature to have is to have names on the containers where the `Services` window is.

I have a `docker-compose.yml` file:

`version: '3.7'

services:
  mongodb:
    image: mongo
    container_name: "db_app"
    ports:
      - "27017:27017"

  frontend:
    image: quizgame/webclient
    container_name: "web_app"
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - "8080:3000"
    environment:
      - NODE_ENV=docker
    depends_on:
      - mongodb`

In `Services` window, there is to running containers.

`running frontend 0.00%
  running mongodb 0.35%`

It should be:

`running frontend (web_app) 0.00%
  running mongodb (db_app) 0.35%`

Love this tool!

**Comment by jesseduffield** [comment]
Sounds like a reasonable request. Given that there if often not much horizontal space, I'm inclined to make this a user configurable thing that defaults to false. Something like:
```
gui:
  showContainerName: true
```
What are your thoughts on that approach?

As for implementation, It would be fairly trivial to write the code, it would just be a matter of changing the GetDisplayStrings() function in `pkg/commands/service.go`. I'm happy to label this as Good First Issue and see if anybody wants to pick it up (I'm happy to lend a hand) :)

**Comment by mudasar187** [comment]
Nice to hear :)

My thoughts is, make it default:

```
gui:
    showContainerName: true
```

And if someone dont want the container name then they can set it to `false`.
If if there is no more space in horizontal then you can show like these examples:

`running thisIsNameOfService (..) 0.00%` (service name is long then add (..) to container name
same on the other way.

And when you click on this service it will show a header on the left windows where `Logs, Stats, Container Config, Top` is and show the container name on the header.

**Comment by pvande** [comment]
My $0.02: I don't really want or need to see the container name in the sidebar (in the header of the main panel would be great though!) **unless** the service has a dynamic container_name.  Knowing what directory lazydocker is running in would be more useful.

My primary use case is in development, where I run a few distinct projects that share service names.  My services have static container names to make linking them together easier, and I'm never running a service as more than a single instance.  If/when I cannot supply a container name (e.g. swarms), I'd still like to see the *service* listed, along with the specific containers.

**Comment by jesseduffield** [comment]
@pvande how about instead of having `lazydocker` in the status panel in the top left, we have the name of the current directory?

As for dynamic container names, I wonder how easy it is to see if a container name is dynamic.

@mudasar187 I'm hesitant to use the ellipsis approach because it seems to me that people either want to know what the container name is or not, and if they have no way to expand the width of the side panels, they can't really access that information. This makes me think we should probably support having a short and long sidepanel view for when you want everything you need in tabular form, but that's a separate issue.

If we could find a way to check if a container name was dynamic, and if so, show it in brackets, would that satisfy both your use cases?

**Comment by mudasar187** [comment]
@jesseduffield Yeah, i think thats a better suggestion :)

**Comment by pvande** [comment]
@jesseduffield I would have suggested putting it in that status panel, but I didn't want to suggest losing the branding!

As for dynamic container names, I'm more or less working from the definition "the docker-compose service does not specify a `container_name` key".  That may not be perfectly adequate, but it seems like a good starting point.

**Comment by pvande** [comment]
Given that Container names may be long, and Services may have multiple containers associated with them, I'm increasingly inclined towards a UI like this:

![image](https://user-images.githubusercontent.com/2350/61998212-95059c80-b061-11e9-9435-d2ccd69f8a52.png)

I would be curious to hear what other people think of this, and how other people would expect it to work.  If there's enough interest in this approach, I may go ahead and try to move it forward myself...

**Comment by kuba-gaj** [comment]
Happy to have a look into this over christmas break if that's ok

**Comment by dawidd6** [comment]
Please, go ahead.

**Comment by mihirsn** [comment]
Hi, is anyone working on it, otherwise I would like to pick this up

**Comment by huyz** [comment]
@loopsFreitag It does seem container names are shown in the general case, but not all. In some cases, I see `ubi8` and I don't know where it comes from, although this discussion thread https://github.com/jesseduffield/lazydocker/discussions/524 suggests it comes from the label block, which I don't see.

**Comment by loopsFreitag** [comment]
@huyz Yes, that is true, today we prioratize taking this information from the container labels, i made a small PR to change this behavior https://github.com/jesseduffield/lazydocker/pull/625

**Comment by huyz** [comment]
@loopsFreitag great, thanks!

**Comment by pasquale95** [comment]
Hi,
why is the PR for this issue not merged yet? I run into this same issue and it would be great to see the container name instead of the `Labels.Name` field.

**Comment by loopsFreitag** [comment]
Seems like this repo is no longer maintained. Since it's not getting updates regularly.
@jesseduffield 

Maybe you can use a custom build

**Comment by huyz** [comment]
what, lazydocker development getting lazy? 😬
